sympy/polys/tests/test_polytools.py

```diff
@@ -3263,7 +3262,7 @@ def test_to_rational_coeffs():
 def test_factor_terms():
     # issue 7067
     assert factor_list(x*(x + y)) == (1, [(x, 1), (x + y, 1)])
-    assert sqf_list(x*(x + y)) == (1, [(x, 1), (x + y, 1)])
```

Comment:
This could be `raises(PolynomialError, lambda: factor_list(x*(x + y)))`.

This error will not appear if one generator is given: `factor_list(x*(x + y), x)` but then `_sorted_factors` will raise `TypeError`. I think that this should not happen. Maybe something like `sorted(facs, key=lambda f: f[1])` could be used. (It only compares the exponents.)

Reply:
Did you mean `raises(PolynomialError, lambda: sqf_list(x*(x + y)))` ?

Reply:
Yes, that is what I meant.